pkg/pipeline/extract/extract_aggregate_test.go

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package extract
+
+import (
+	jsoniter "github.com/json-iterator/go"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/extract/aggregate"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+	"testing"
+)
+
+func Test_Extract(t *testing.T) {
+	yamlConfig := `
+aggregates:
+- name: bandwidth
+  by:
+  - service
+  operation: sum
+  recordkey: bytes
+- name: bandwidth_count
+  by:
+  - service
+  operation: count
+  recordkey: ""
+`
+	var err error
+	yamlData := make(map[string]interface{})
+	err = yaml.Unmarshal([]byte(yamlConfig), &yamlData)
+	require.NoError(t, err)
+	var json = jsoniter.ConfigCompatibleWithStandardLibrary
+	jsonBytes, err := json.Marshal(yamlData["aggregates"])
+	require.NoError(t, err)
+	config.Opt.PipeLine.Extract.Aggregates = string(jsonBytes)
+
+	extractAggregate, err := NewExtractAggregate()
+	require.NoError(t, err)
+
+	input1 := []config.GenericMap{
+		{"service": "http", "bytes": 10.0},
+		{"service": "http", "bytes": 20.0},
+		{"service": "tcp", "bytes": 1.0},
```

Comment:
Why are "bytes" floats and not integers?

Reply:
I'll fix this